panel/widgets/chat.py

```diff
@@ -0,0 +1,1175 @@
+                {% else %}
+                src="${_icon_base_url}{{ icon_name }}.svg"
+                {% endif %}
+                style="width: ${icon_size}; height: ${icon_size}; cursor: pointer;"
+                onclick="${script('update_value')}">
+            </img>
+            {% endfor %}
+        </div>
+    """
+
+    _scripts = {
+        "update_value": """
+            const reaction = event.target.alt;
+            const icon_name = data.options[reaction];
+
+            let src;
+            if (data.value.includes(reaction)) {
+                src = `${data._icon_base_url}${icon_name}.svg`;
+                data.value = data.value.filter(r => r !== reaction);
+            } else {
+                src = reaction in data.active_icons
+                    ? `${data._icon_base_url}${data.active_icons[reaction]}.svg`
+                    : `${data._icon_base_url}${icon_name}-filled.svg`;
+                data.value = [...data.value, reaction];
+            }
+            event.target.src = src;
+        """
+    }
+
+    _stylesheets: ClassVar[List[str]] = [
+        f"{CDN_DIST}css/chat_reaction_icons.css"
+    ]
+
+    def __init__(self, **params):
+        super().__init__(**params)
+        if self.icon_size.endswith("px"):
+            self.max_width = int(self.icon_size[:-2])
+
+
+class ChatEntry(CompositeWidget):
+    """
+    A widget for displaying chat messages with support for various content types.
+
+    This widget provides a structured view of chat messages, including features like:
+    - Displaying user avatars, which can be text, emoji, or images.
+    - Showing the user's name.
+    - Displaying the message timestamp in a customizable format.
+    - Associating reactions with messages and mapping them to icons.
+    - Rendering various content types including text, images, audio, video, and more.
+
+    Parameters
+    ----------
+    value : object
+        The message contents. Can be a string, pane, widget, layout, etc.
+    user : str
+        Name of the user who sent the message.
+    avatar : str | BinaryIO
+        The avatar to use for the user. Can be a single character text, an emoji,
+        or anything supported by `pn.pane.Image`. If not set, uses the
+        first character of the name.
+    reactions : List
+        Reactions to associate with the message.
+    reaction_icons : ChatReactionIcons | dict
+        A mapping of reactions to their reaction icons; if not provided
+        defaults to `{"favorite": "heart"}`.
+    timestamp : datetime
+        Timestamp of the message. Defaults to the instantiation time.
+    timestamp_format : str
+        The timestamp format.
+    show_avatar : bool
+        Whether to display the avatar of the user.
+    show_user : bool
+        Whether to display the name of the user.
+    show_timestamp : bool
+        Whether to display the timestamp of the message.
+    """
+
+    value = param.ClassSelector(class_=object, doc="""
```

Comment:
To me it seems like the `ChatEntry.value` is not being set? Something like the below fails.

```python
@pytest.fixture
def chat_feed():
    return ChatFeed()

def test_can_provide_response(chat_feed):
    """The user can quickly add a new response"""
    prompt = "What is 2+2"
    response = get_response(prompt)
    # When: this is our recommended best practices
    entry = chat_feed.send(value = response, user="Assistant", )
    # Then
    chat_feed.entries==[entry]
    assert len(chat_feed.value)==1    
```

Reply:
If its on purpose I would recommend using the parameter name `value` instead of the parameter `entries` because we normally use either `value` or `object`.

Reply:
I removed `entries` and bidirectionally `bind`ed `objects` to `value`.